src/endpointsApi.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+const logger = require('./logger')
+const EndpointModel = require('./models/endpoints')
+const {handleServerError, validateEndpoint} = require('./util')
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const saveEndPoint = body => {
+  return new Promise((resolve, reject) => {
+    const endpoint = new EndpointModel(body)
+    endpoint.save(err => {
+      if (err) reject(err)
+      resolve(endpoint)
+    })
+  })
+}
+
+const updateEndPoint = (endpointId, body) => {
+  return new Promise((resolve, reject) => {
+    EndpointModel.findOneAndUpdate(
+      {_id: endpointId},
+      body,
+      {new: true, runValidators: true},
+      (err, result) => {
+        if (err) reject(err)
+        resolve(result)
+      }
+    )
+  })
+}
```

Comment:
@bradsawadye I think this code needs to be moved into a separate file - probably `db.js`. This code is very specific to Mongoose and therefore Mongo. This makes our Mediator code tightly coupled to the DB framework. If we update Mongoose or Mongo and there are API changes it would mean having to look through all our mediator code to possibly update our DB related code.
If we move DB specific code into its own file, we separate concerns and we would only have to update one file when there are DB updates.
On the OHIE-374 branch, these functions are in the `db.js` file.
@BMartinos what do you think?

Reply:
@MattyJ007 agreed. Lets move this into a db specific file that handles the db related requests. It could then be easily swapped out for a different db and not affect the actual business logic related to the CRUD endpoints

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/48/commits/5ca6f708987ff58c334a9cd9fca4b39f146562a6